KILTprotocol/sdk-js
# ci: improvements for scheduled workflow

## no ticket

Just some smaller fixes on the scheduled workflow.

1. Update the spiritnet metadata before running tests. This ensures that type encoding and other assumptions still match what is going on on the mainnet.
2. Small fix to improve caching
3. Upload `yarn.lock` to provide a resource for investigation if build fails
4. Commit latest version of spiritnet metadata so that all other tests are representative of current chain state.

## How to test:

See [run](https://github.com/KILTprotocol/sdk-js/actions/runs/2022027259) triggered by [cafe05f](https://github.com/KILTprotocol/sdk-js/pull/507/commits/cafe05fa45e896ca6e14c6edb1db02e93dc23b8f)


## Checklist:
